### PR TITLE
feat: Add ability to push to multiple registries

### LINKED ISSUE #218: Feature Request: Push to multiple registries
It would be nice to be able to do something like this:
```yaml
kind: pipeline
name: default

steps:
  - name: docker
    image: thegeeklab/drone-docker-buildx:23
    privileged: true
    settings:
      repo-list:
        - username: octocat
          password: ****
          repo: octocat/example
        - registry: ghcr.io
          username: gh-octocat
          password: ****
          repo: ghcr.io/gh-octocat/example
        - registry: quay.io
          username: quay-octocat
          password: ****
          repo: quay.io/quay-octocat/example
          ...
```

### PR DESCRIPTION
**Pull request by Maks1mS**
Resolves #218.

How I tested it:
```sh
$ USERNAME="maks1ms"
$ IMAGENAME="hello-world"
$ DOCKER_REGISTRY_PASSWORD=...
$ GITHUB_REGISTRY_PASSWORD=...

$ PLUGIN_REGISTRIES=`cat <<EOF
registries:
  - username: "$USERNAME"
    password: "$DOCKER_REGISTRY_PASSWORD"
  - registry: "ghcr.io"
    username: "$USERNAME"
    password: "$GITHUB_REGISTRY_PASSWORD"
EOF
`

$ docker run --rm \
  -e PLUGIN_TAG=latest,2 \
  -e PLUGIN_REPO=$USERNAME/$IMAGENAME,ghcr.io/$USERNAME/$IMAGENAME \
  -e PLUGIN_REGISTRIES="$PLUGIN_REGISTRIES" \
  -e PLUGIN_DOCKERFILE="Dockerfile.multiarch" \
  -e DRONE_COMMIT_SHA=00000000 \
  -v $(pwd):$(pwd) \
  -w $(pwd) \
  --privileged \
  thegeeklab/drone-docker-buildx
```

### CODE REVIEW

**Comment by xoxys** [comment]
Thanks! As `repo` is a StringSlice now `repo: octocat/example` doesn't work any more right?

File: _docs/data/data.yaml [hunk 1]
```diff
@@ -293,3 +293,36 @@ properties:
       This should be used with caution and avoided whenever possible.
     type: list
     required: false
+
+  - name: registries
+    description: |
+      Credentials for registries.
+      Example:
```
Comment (xoxys) [inline _docs/data/data.yaml:300]: I would not include the example twice. If we add it to the `Examples` section already, that should be enough.

File: plugin/impl.go [hunk 2]
```diff
@@ -185,6 +217,8 @@ func (p *Plugin) Execute() error {
 	switch {
 	case p.settings.Login.Password != "":
 		logrus.Info("Detected registry credentials")
+	case p.settings.Login.RegistriesYaml != "":
+		logrus.Info("Detected multiple registries credentials")
```
Comment (xoxys) [inline plugin/impl.go:221]:
```suggestion
		logrus.Info("Detected multiple registry credentials")
```

Just a small nitpick

**Review by xoxys — CHANGES_REQUESTED** [review]

File: _docs/data/data.yaml (hunk 1 above)
Comment (Maks1mS) [reply _docs/data/data.yaml]: 7f7cff265e2d05004209aedcbd848a8ac19fde1a

File: plugin/impl.go (hunk 2 above)
Comment (Maks1mS) [reply plugin/impl.go]: f93d34038d1f5810a9b1d85fd6a7cb3114097fca

**Comment by Maks1mS** [comment]
> Thanks! As `repo` is a StringSlice now `repo: octocat/example` doesn't work any more right?

As I understand it, it will work, because there is no difference between a list of one element and just a value.
For reliability, I will now check it myself and write the result.

**Comment by Maks1mS** [comment]
Yep, it works.

**Comment by xoxys** [comment]
Thanks for testing.

**Review by xoxys — APPROVED** [review]

**Pull request merged**